src/audio/buffer.c

```diff
@@ -272,8 +272,8 @@ void buffer_free(struct comp_buffer *buffer)
 	/* In case some listeners didn't unregister from buffer's callbacks */
 	notifier_unregister_all(NULL, buffer);
 #if CONFIG_ZEPHYR_DP_SCHEDULER
-	ring_buffer_free(buffer->stream.dp_queue_sink);
-	ring_buffer_free(buffer->stream.dp_queue_source);
+	ring_buffer_free_legacy(buffer->stream.dp_queue_sink);
```

Comment:
this "legacy" is introduced ONLY to make this commit compile (and problably work). It is removed in the very next commit